### DISCUSSION TITLE
SPRINT 1- finished

### DISCUSSION THREAD

**Discussion by NicoSjokk** [IT2901Nordic2024/project#7] [General]
🎉🎉 Sprint 1 completed! 🎉🎉

Last week we accomplished the sprint goal for sprint 1 which was to establish a proper connection between firmware, backend and frontend. 
The code now sends uptime from the devices to AWS, which saves the data stored as shadows to the database and the frontend can pull the current data using a REST api. 

We've just had our sprint planning meeting for sprint 2 where our goal is to establish user authentication in the backend and frontend as well as configure the firmware the AWS IoT sample to use the sensor data to send the payload based on a sensor event instead of time.

Attached are screenshots of the frontend user interface:

<img src="https://github.com/IT2901Nordic2024/project/assets/114810966/d7a0ac6f-6606-4099-a65c-45614946dc00" height="550">

<img src="https://github.com/IT2901Nordic2024/project/assets/114810966/b2c38527-1196-44a9-ba16-5918d68e02d4" height="550">

<img src="https://github.com/IT2901Nordic2024/project/assets/114810966/547bd477-4a07-4688-a02b-17b8f784bad6" height="550">

<img src="https://github.com/IT2901Nordic2024/project/assets/114810966/a02075f1-f31c-4bb8-b0fd-95f0f8373a34" height="550">

Additionally, here are the issues we have created for sprint 2:

![IssuesSprint2](https://github.com/IT2901Nordic2024/project/assets/114810966/bb8f8992-b45f-478c-93eb-c8440c915790)

Do you agree with the sprint goal and issues for sprint 2 or is there something you would rather we focus on instead?

**Comment by coderbyheart** [comment]
There is a lot of "technical ceremony" in the plan, while at the same time there is no mention of the user and what feature all of this supports **at all**, and why this is important for the sprint. 

None of these are mentioned: tracking user behaviour, counting cups of coffee, or minutes exercised.

In this early phase of the project you need to focus on building a working prototype of the final product. User registration for example is completely irrelevant right now.

As a product owner I would say: "Show me how to log in on a website!" but "Show me how the nRFHabitTracker works!"

**Reply by coderbyheart** [reply]
Another way to figure out what's important:

"What are the features that are UNIQUE to this project?"

Focus on those. They are hard.

"What does every app have?"

These can be added later. They are easy, because they have been done before.

**Reply by VebjornNyvoll** [reply]
Thanks for your input! We agree and have come up with new goals for sprint 2. We had already based our issues on user stories, but completely agree we should've sent those.

Our firmware team will continue working on the same issues as they're relevant to the following user story:
- As a user I want to track an activity exclusively using my Thingy91, because I want to be able to track the activities I have configured using the web interface.

Our frontend and backend team will focus on the following user stories:
- As a user I want to see a timeline of my tracked activities, because I want to track my habits.
- As a user I want to configure the device using a web interface, because I want to be able to choose what habits I am tracking.

To show how this flow will work for you as a customer, we will develop an emulator for the firmware that allows tracking of configured activities, for instance how many coffees a user is drinking.

Hopefully this is more in line with your expectations and feel free to give input on the changes. :)

**Reply by coderbyheart** [reply]
I think you should spend some time on defining these:

- (configured|tracked) activity
- (tracked) habit
- timeline

and come up with some mockups for concrete examples that explain how that is supposed to work.

**Comment by VebjornNyvoll** [comment]
Thank you for the feedback! :)

After defining the terms we have decided on moving away from the term activity and will instead use the term habit. Previously we meant the term activity as a user-configured event to track and a habit would be that tracked activity over time. We believe using the term activity for what is being tracked and a habit as an activity over-time is not immediately clear to the user compared to using habit from the get-go and that is why we have now made this change.

We have updated the [README.md for the project repository](https://github.com/IT2901Nordic2024/project) with a more detailed write-up of the key term definitions. This allows us to further expand it with new definitions later if needed and keeps them all in one place.

**Reply by coderbyheart** [reply]
This is a good start. Building on this you should go through the exercise of defining, how a habit of `type`: `counter` will be used to track push-ups. Explain this as precise as possible: what will the user need to do in order to track the number of push-ups they are doing?
